mlir/lib/Dialect/XeGPU/Transforms/XeGPUFoldAliasOps.cpp

```diff
@@ -48,15 +50,51 @@ LogicalResult XegpuCreateNdDescOpSubViewOpFolder::matchAndRewrite(
     return rewriter.notifyMatchFailure(descOp, "not a subview producer");
   if (!subViewOp.hasUnitStride())
     return rewriter.notifyMatchFailure(descOp, "requires unit strides");
+  if (!subViewOp.getSource().getType().hasStaticShape())
+    return rewriter.notifyMatchFailure(descOp, "requires static shape");
```

Comment:
Is this restriction necessary?
I think nd-descriptor supports dynamically shaped source.